### PR TITLE
feat: add developer state protection from payload reset

### PR DESCRIPTION
**Pull request by sidnov**
Fixes #[issue_no]

## Summary of Changes

When A32NX_DEVELOPER_STATE is true, MCDU reloads will not affect payload or pax values.

## Additional context

Discord username (if different from GitHub): sidnov#8337

## Testing instructions

Developer-side changes, no QA needed.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_Boarding.js [hunk 1]
```diff
@@ -19,11 +19,14 @@ class A32NX_Boarding {
...
-        await this.loadCargoZero();
-        await this.loadCargoPayload();
+        const inDeveloperState = SimVar.GetSimVarValue("L:A32NX_DEVELOPER_STATE", "Bool");
+        if (!inDeveloperState) {
+            // Set default pax (0)
+            this.setPax(0);
+            this.loadPaxPayload();
+            this.loadCargoZero();
+            this.loadCargoPayload();
+        }
```
Comment (davidwalschots) [inline flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_Boarding.js:29]: **thought**: There used to be lots of `await` usages there. Now there are none. That _might_ be fine. But then probably you also want to remove the `async` modifier on the function?

**Review by davidwalschots — COMMENTED** [review]
Could you explain why specifically we'd not want this in this situation?

File: flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_Boarding.js (hunk 1 above)
Comment (sidnov) [reply flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_Boarding.js:29]: This case (reset payload to 0) simply sets the values, while some other usages of the same fn would ideally require an ``await`` before proceeding to cargo loading

Comment (davidwalschots) [reply flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_Boarding.js:29]: I'm not quite sure what you mean. As there are no `await`s, one can remove `async` from `async init() {`.

Comment (davidwalschots) [reply flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_Boarding.js:29]: I'd definitely keep `await` in when the function exposes a Promise (is async). You shouldn't consdier the internals of the functions from the call site, as those might change in the future.

Comment (sidnov) [reply flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_Boarding.js:29]: I think I misunderstood from the first comment about async, I get what you meant 👍🏻

**Review by tracernz — APPROVED** [review]
Legend! 😁

**Comment by tracernz** [comment]
> Could you explain why specifically we'd not want this in this situation?

When developing e.g. the FMS... the payload gets reset to zero each time you reload the FMGC+MCDU instrument prior to this PR.

**Pull request merged**